src/app/shared/form/builder/ds-dynamic-form-ui/models/empty-select/dynamic-empty-select.component.ts

```diff
@@ -22,131 +22,148 @@ import { isEmpty } from 'src/app/shared/empty.util';
-                }
-
-                // Update options
-                // Retrieve the "options" key from the "relevantData" object
-                let { options } = relevantData;
-                this.optionsList = options.map((option) => this.generateVocabularyEntry(option));
+    this.subscriptions.add(
+      this.formDynamicUpdateService.dynamicFieldUpdateEvent
+        .subscribe((data: any) => {
+          let relevantData = data['data-' + replaceAll(this.model.id, '_', '.')];
+          // If there is relevant data for the current field
+          if (relevantData) {
+            // Update value if given
+            if (relevantData.value) {
+              let remoteValue = this.generateVocabularyEntry(relevantData.value);
+              this.selectOption(remoteValue);
             }
-        }));
-    }
-
-    /**
-     * Converts an option coming from the backend into a VocabularyEntry.
-     * @param option The option to generate the VocabularyEntry from.
-     * @returns The generated VocabularyEntry.
-     */
-    generateVocabularyEntry(option: {value: string, displayed: string}): VocabularyEntry {
-        return Object.assign(new VocabularyEntry(), { display: option.displayed, value: option.value });
-    }
-
-    /**
-     * Set the current value with the given value as an observable.
-     * @param value The value to set.
-     */
-    setCurrentValue(value: any) {
-        let result: Observable<string>;
-        if (isEmpty(value)){
-            result = observableOf('');
-        } else if (typeof value === 'string'){
-            result = observableOf(value);
-        }
-        this.currentValue = result;
-    }
-
-    /**
-     * Converts an item from the result list to a `string` to display in the `<input>` field.
-     */
-    inputFormatter = (x: VocabularyEntry): string => x.display || x.value;
-
-    /**
-     * Opens dropdown menu
-     * @param sdRef The reference of the NgbDropdown.
-     */
-    openDropdown(sdRef: NgbDropdown) {
-        if (!this.model.readOnly) {
-            sdRef.open();
-        }
-    }
-
-    /**
-     * KeyDown handler to allow toggling the dropdown via keyboard
-     * @param event KeyboardEvent
-     * @param sdRef The reference of the NgbDropdown.
-     */
-    selectOnKeyDown(event: KeyboardEvent, sdRef: NgbDropdown) {
-        const keyName = event.key;
-
-        if (keyName === ' ' || keyName === 'Enter') {
-            event.preventDefault();
-            event.stopPropagation();
-            sdRef.toggle();
-        } else if (keyName === 'ArrowDown' || keyName === 'ArrowUp') {
-            this.openDropdown(sdRef);
-        }
+            // Update options
+            //   * Retrieve the "options" key from the "relevantData" object
+            //   * If only one option is possible, select it
+            this.optionsList = relevantData.options.map((o) => this.generateVocabularyEntry(o));
+            if (this.optionsList.length == 1) {
```

Comment:
My IDE is showing a warning here stating that it should be '===' (Checking value and type) rather than '==' (checking only the value);

Reply:
Your IDE is very safe :-) Here I use a generic javascript builtin attribute that will return "number", so I can reasonably use "=="